Functions/External/xASL_ext_VABYRun.m

```diff
@@ -0,0 +1,90 @@
+function [x, Result1] = xASL_ext_VABYRun(VABYCommand, x, NicenessValue, bVerbose)
+%xASL_ext_VABYRun Run VABY from Matlab (ExploreASL)
+%
+% FORMAT: [x] = xASL_adm_RunVABY(VABYCommand, x[, NicenessValue, bVerbose])
+%
+% INPUT:
+%   VABYCommand     - Command line job for VABY (REQUIRED)
+%   x               - structure containing fields with all information required to run this submodule (REQUIRED)
+%   NicenessValue   - the linux nice parameter, a scale with 40 integers 
+%                     as index of the priority granted to a process. Lower
+%                     = higher priority, higher = lower priority. If no
+%                     other processes are running, a lower priority will
+%                     still use many resources.
+%                     Provide a number between [-20 +19], (OPTIONAL,
+%                     DEFAULT=10)
+%   bVerbose        - verbose output (OPTIONAL, DEFAULT true)
+%
+% OUTPUT:
+%   x               - as input, outputting VABY dir (OPTIONAL)
+%   Result1         - Result1 describes if the execution was successful
+%                     (0 = successful, NaN = no VABY found, 1 or other = something failed)
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function runs an VABY command from ExploreASL:
+%
+%              1. Checking the VABY dir
+%              2. Running the command
+%
+% Supports .nii & .nii.gz, Linux, MacOS & Windows
+% 
+% EXAMPLE: xASL_ext_VABYRun(VABYCommand, x);
+% __________________________________
+% Copyright (C) 2015-2024 ExploreASL
+% Licensed under Apache 2.0, see permissions and limitations at
+% https://github.com/ExploreASL/ExploreASL/blob/main/LICENSE
+% you may only use this file in compliance with the License.
+% __________________________________
+
+
+
+%% Admin
+if nargin<2 || isempty(x)
+    warning('x input missing');
+    x = struct;
+end
+if nargin<3 || isempty(NicenessValue)
+    NicenessValue = 10;
+end
+if nargin<4 || isempty(bVerbose)
+    bVerbose = true;
+end
+
+% Defaults
+Result1 = NaN;
+
+%% Find VABY directory
+[VABYdir, x] = xASL_ext_VABYSetDir(x);
+
+if min(isnan(VABYdir))
+    % Script will return Result1=NaN to show that there is no VABY
+    % installation found
+    warning('No VABY installation found, skipping VABY function');
+    return;
+end
+
+% Setup VABY quantification
+VABYinit = '';
+VABYoutput = '';
+OutputString = '.nii';
+if strcmp(VABYCommand(1:4),'vaby')
```

Comment:
Sorry, VABYinit and VABYoutput should be removed. Nice catch. That's a copy paste from FSL, which is not necessary from VABY so it was set empty. With separate functions, we should of course remove it...

You either provide full path to "vaby" or just the command and the path is added here. Same as for FSL. You always need VABY folder. It's just that you can provide it yourself to the function. So this is OK. I added a comment about this. Done in 15cf7ec9

It should be strcmp, because the command "vaby" is provided by us.

This is not about path to data but path to the command, so 1:4 is robust...

Reply:
I don't get it. The comment says "check correct path" but you now say vaby command. How can a correct path start with 'vaby'? A correct path is e.g. /opt/vaby

or something similar

Also note: there are users that have:

/vaby/RunExploreASL/data/...
/vaby/RunExploreASL/vabycode/

OK now I get it.
So it should be:

if strcmp(VABYCommand(1:4),'vaby') ~exist(VABYCommand, 'dir')

right?

Reply:
Yes - it's checking if VABYcommand has the command without path or with path ("vaby file.nii' or "/usr/local/vaby1.2/vaby file.nii"). I adapted the comment and code..

Also - we don't need to add the "exist" command. Because we already have a check on directory existence above